cabal-install/src/Distribution/Client/TargetSelector.hs

```diff
@@ -1763,14 +1763,15 @@ getKnownTargets dirActions@DirActions{..} pkgs = do
       knownComponentsOther   = allComponentsIn opinfo
     }
   where
-    selectPrimaryPackage :: FilePath
-                         -> [KnownPackage]
-                         -> ([KnownPackage], [KnownPackage])
-    selectPrimaryPackage cwd = partition isPkgDirCwd
-      where
-        isPkgDirCwd KnownPackage { pinfoDirectory = Just (dir,_) }
-          | dir == cwd = True
-        isPkgDirCwd _  = False
+    selectPrimaryPackage _ [] = return ([] , []) 
```

Comment:
Type signature here wouldn't hurt. Add `forall a.` to the type signature above it make it possible. And tnen `ScopedTypeVariables` to file extensions at the top, if it yells at you.